compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/peel_for_loop.mlir

```diff
@@ -1,6 +1,7 @@
 // RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-amdaie-peel-for-loop{peeling-type=first}))"  --split-input-file %s | FileCheck %s --check-prefix=PEEL-FIRST
 // RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-amdaie-peel-for-loop{peeling-type=last}))"  --split-input-file %s | FileCheck %s --check-prefix=PEEL-LAST
 // RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-amdaie-peel-for-loop{peeling-type=first-last}))"  --split-input-file %s | FileCheck %s --check-prefix=FIRST-LAST
+// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-amdaie-peel-for-loop{peeling-type=first}))"  --split-input-file %s | FileCheck %s --check-prefix=OVERWRITE-FLAG
```

Comment:
I think this could be confusing allowing a user specified peeling-type to be overwritten like this. If there was peeling-type=auto, or peeling-type=unset, changing those to another peeling-type would make sense .

Reply:
So the default peelingType is set as `PeelingType::First` at https://github.com/nod-ai/iree-amd-aie/blob/main/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.td#L243 because it is the peeling function we want to use for GEMM. And this flag is equivalent to `--pass-pipeline="builtin.module(func.func(iree-amdaie-peel-for-loop))"`, and it is overwritten when it detects the matmul-elementwise pattern.

Reply:
Thanks for your suggestion. Now I change the default setting in td file as "None", and in implementation call `peelForLoopFirstIteration` by default. In this way, for normal GEMM it will peel the first iteration automatically, and for matmul-elementwise case it will first assign the PeelingType as "FirstLast".